Peer Review Mendix Cloud (#7489)
mendix/docs: content/en/docs/developerportal/deploy/mendix-cloud-deploy/_index.md

--- 
title: "Mendix Cloud" 
url: /developerportal/deploy/mendix-cloud-deploy/ 
category: "Deployment" 
weight: 20 
description: "Describes how to deploy to Mendix Cloud." 
tags: ["Deploy","Mendix Cloud","Developer Portal", "Free App", "licensed", "limitations", "Mendix Cloud Dedicated"] 
aliases: 
    - /developerportal/howto/deploying-to-the-cloud.html 
    - /mendixcloud/deploying-to-the-cloud.html 
    - /developerportal/howto/deploying-to-the-cloud 
    - /mendixcloud/deploying-to-the-cloud 
#To update these screenshots, you can log in with credentials detailed in How to Update Screenshots Using Team Apps. 
#If moving or renaming this doc file, implement a temporary redirect and let the respective team know they should update the URL in the product. See Mapping to Products for more details. 
#The anchor #plans, below, is mapped from Control Center within the Developer Portal. 
--- 
 
## 1 Introduction 
 

> Mendix Cloud is a public cloud service for Mendix applications; it is the default deployment option for Mendix applications. You can either deploy a limited [Free App](https://www.mendix.com/evaluation-guide/evaluation-learning/getting-started/#evaluate-before) or an app running on a licensed cloud node. 

Author: Should we have a semi-colon here. I tend to try to write around this, but it is grammatically correct.

> Organizations can also have their own Mendix Cloud. This service is named [Mendix Cloud Dedicated](https://www.mendix.com/evaluation-guide/app-lifecycle/mendix-cloud-overview/#mendix-cloud-vpc). Mendix Cloud Dedicated is a single-tenant instance of the public Mendix Cloud, running apps of only one organization. 

Author: I think a "the" is left over from the change to just call it Mendix Cloud. Doesn't fit an easy regular expression pattern. I've suggested a new place to put it.
```suggestion
Organizations can also have their own Mendix Cloud. This service is named [Mendix Cloud Dedicated](https://www.mendix.com/evaluation-guide/app-lifecycle/mendix-cloud-overview/#mendix-cloud-vpc). Mendix Cloud Dedicated is a single-tenant instance of public Mendix Cloud, running the apps of only one organization. 
```

>  

Author: I wonder if we should say a bit more about what Mendix Cloud is technically (e.g. that it runs on AWS, has various regions, etc.

### 1.1 Free App {#free-app} 
 

> If you are new to the Mendix community and would like to deploy and share your own app, you can do so for free on Mendix's public cloud offering. The Free App environment allows any Mendix developer to create and share their applications with their users. Note that a Free App does not support complex or large applications. 

Author: Not sure about "Mendix's public cloud offering". This can also apply to the licensed nodes. But don't know if it is worth coming up with something else.

I think it is "a" free app environment?
```suggestion
If you are new to the Mendix community and would like to deploy and share your own app, you can do so for free on Mendix's public cloud offering. A Free App environment allows any Mendix developer to create and share their applications with their users. Note that a Free App does not support complex or large applications. 
```

 
{{% alert color="info" %}} 
Free Apps are part of Mendix's Free Edition. 
 
If you are not currently a customer but would prefer to use a licensed cloud node, you can find more information on the [Mendix Platform Pricing](http://www.mendix.com/pricing) page. 
 

> If you are an existing customer, you should deploy into your licensed cloud node. 

Author: into -> onto.
Never sure about deployment prepositions.

| Custom domains | Not available | Can be configured from the Developer Portal | 
| Access restriction profiles | Not available | Can be configured from the Developer Portal | 
| Client certificates | Not available | Can be configured from the Developer Portal | 
| Starting and stopping your app manually | Not available | Available in the Developer Portal | 
| Constants | Defined in Studio Pro | Configurable through environment variables in the Developer Portal | 
| Runtime settings | Not available | Configurable through runtime and settings in the Developer Portal | 
| Scalability | Only one instance and a fixed amount (1 GiB) of memory | Configurable in the Developer Portal | 
| Metrics, alerts, and log levels | Not available | Available | 
| Historic app logs | Not available; only live logs are available | Available | 
| Backups | Performed daily and cannot be triggered manually; stored for up to two weeks | Performed daily and can also be created manually; kept for up to one year, depending on your plan | 
| Support | No support | Depends on your license option | 
 

> ¹ End-users of your app are classified as either internal or external. You will need to report this for licensing purposes using either the [USAGE_METRICS_EMAIL_FIELDS custom variable](/developerportal/deploy/environments-details/#custom-environment-variables) (if you are using email domains to distinguish between them) or by [populating the user type](/howto/monitoring-troubleshooting/populate-user-type/) for each end-user of your app. Only end-users whose Mendix accounts are marked as **Active** are counted towards the number of end-users of the app. 

Author: It feels wrong to have this in the **Free Apps** section. It is something you need to do for a licensed app. I wonder if it can go elsewhere?

 
#### 1.1.1 Sleep Mode 
 
As noted in the table above, a Free App goes into sleep mode after about an hour of inactivity. If you access the app while it is inactive, it displays the following message:  
 
{{< figure src="/attachments/developerportal/deploy/mendix-cloud-deploy/appresumed.png" alt="A Resuming app message" width=60% >}} 
 
If, after a couple of minutes, your app does not wake up, contact [Mendix Support](http://support.mendix.com). 
 

> You can upgrade a Free App to a licensed node with a node in Mendix Cloud. For more information, see [Licensing Mendix Cloud Apps](/developerportal/deploy/licensing-apps/). 

Author: I think probably:
```suggestion
You can upgrade a Free App to a licensed app running on a node in Mendix Cloud. For more information, see [Licensing Mendix Cloud Apps](/developerportal/deploy/licensing-apps/). 
```

 
If you need to use a different database, consider deploying your app to a different platform. For more information, see [Deployment](/developerportal/deploy/). 
 
### 1.5 URLs and Ports 
 
Apps that run on Mendix Cloud are automatically given their own URLs. The format of the URL depends on the license and environment type, and can be one of the following: 
 
| License Type | Environment | URL Format | Example URL | 
| ------------ | ----------- | ---------- | ----------- | 
| Licensed app | Production  | Depends on the region:<br /> `{app-name}.mendixcloud.com`<br />or<br />`{app-name}.apps.{region}.mendixcloud.com` | `myappname.mendixcloud.com`, <br /> `myappname.apps.ap-3a.mendixcloud.com` | 
| Licensed app | Test, acceptance, flexible environments | Depends on the region:<br /> `{app-name}-{environment-type}.mendixcloud.com`<br />or<br />`{app-name}-{environment-type}.apps.{region}.mendixcloud.com` | `myappname-accp.mendixcloud.com`, <br /> `myappname-accp.apps.ap-3a.mendixcloud.com` | 
| Free App     | N/A         | `{app-name}-sandbox.mxapps.io`<br />or<br />`{app-name}.mxapps.io` | `myfreeappname.mxapps.io` | 
 

> {{% alert color="info" %}}As shown in the table above, the URL of an app on some clusters in a region contains `apps.{region}`. You can select the region for only your app, not for the cluster.{{% /alert %}} 

Author: > You can select the region for only your app, not for the cluster.

I don't really understand what we are saying here - mainly because I don't think we really explain what a cluster is and haven't explained that we are running on AWS and all the other background which would help to make this clear.

 
You can customize a URL by adding [custom domains](/developerportal/deploy/custom-domains/). 
 
Mendix apps cannot use custom ports. They communicate on the standard HTTP and HTTPS ports (`80` and `443`); connections to HTTP (`80`) are redirected to HTTPS (`443`). 
 

> ## 2 Deploying an App to Mendix Cloud {#deploy-app-mendix-cloud} 

Author: I agree with the suggestion of moving it to a separate document. This document is long because of the various tables, so pulling this out would make it more focused.

 
The method for deploying an app to Mendix Cloud differs depending on whether you have a licensed app or a Free App: 
 
* For a licensed app, there are several deployment options: 
    * Manual deployment – Create a deployment package via Studio Pro or the Developer Portal, and then deploy the package to a node environment where you can run it. For details on how to do this, see [Deploying a Licensed App](#deploy-licensed-app), below. This is the simplest deployment method, especially if you are just getting started with Mendix. 

>     * Pipelines built with low code – If you want to automate your build and deployment process, you can use Pipelines in the Developer Portal to design and activate a pipeline with a set of configurable, low-code steps. For details, see [Pipelines (Beta)](/developerportal/deploy/pipelines/). The Pipelines feature is intended to make it quick and easy for teams to automate their CI/CD process, without requiring any third-party tools or DevOps expertise. 

Author: I think I'd rather we didn't say Beta here - I think people can find out it is a beta when they get to the linked page. So perhaps move the link and remove that?

And be a bit more positive about describing the feature?
```suggestion
    * Pipelines built with low code – If you want to automate your build and deployment process, you can use  [Pipelines](/developerportal/deploy/pipelines/) in the Developer Portal to design and activate a pipeline with a set of configurable, low-code steps. The Pipelines feature makes it quick and easy for teams to automate their CI/CD process, without requiring any third-party tools or DevOps expertise. 
```

    * Pipelines built with APIs – If you use Jenkins, GitLab, or another CI/CD tool, you can use Mendix APIs to automate your build and deployment process. For details, see [Implement a Simple CI/CD Pipeline with Mendix APIs](/howto/integration/implement-cicd-pipeline/). This method is highly customizable but does require CI/CD tooling and DevOps expertise to set up. 
* For a Free App, you deploy the app directly from Studio Pro. For details on how to do this, see [Deploying a Free App](#deploy-free-app), below. 
 
{{% alert color="info" %}} 
For a hands-on introduction to the different deployment options for licensed apps, try the learning path [Choose the Right Software Delivery Approach](https://academy.mendix.com/link/paths/156/Choose-the-Right-Software-Delivery-Approach). 
{{% /alert %}} 
 
### 2.1 Deploying a Licensed App{#deploy-licensed-app} 

>  

Author: I think it might be good to summarize the process and introduce that there are two stages we are going to describe: **create package** and **deploy**

#### 2.1.1 Prerequisites 
 
Before starting the process for deploying a licensed app, make sure to complete these prerequisites: 
 
* Your app is linked to a licensed cloud node 
* You have [transport rights](/developerportal/deploy/node-permissions/#transport-rights) 
* Your [two-factor authentication](/developerportal/deploy/two-factor-authentication/) is set up 
* The deployment package for your app is not bigger than 1 GB (uncompressed) 
 
{{% alert color="info" %}}You can verify that your app is under the 1 GB limit by creating a deployment package, as described below, and then viewing that package's size in the package details.{{% /alert %}} 
 
#### 2.1.2 Creating a Deployment Package for a Licensed App 
 

> There are two methods for creating a deployment package on Mendix Cloud. You can create a deployment package directly via Studio Pro, or you can create a package via the Team Server through the Developer Portal. 

Author: My eyes get tired reading "a deployment package". Perhaps the reader would understand "it" in a couple of places? And perhaps explain where the Team Server comes into it.
```suggestion
There are two methods for creating a deployment package on Mendix Cloud. You can create it directly via Studio Pro, or through the Developer Portal, using a model committed to the Team Server. 
```

 
##### 2.1.2.1 Creating a Deployment Package via Studio Pro 
 
To create a deployment package directly from Studio Pro, follow these steps: 
 
1. Open [Studio Pro](http://appstore.home.mendix.com/link/modeler/). 
2. Open the licensed app. 
3. In the top menu bar, click **App** > **Deploy to Licensed Cloud Node**. 
 
    {{< figure src="/attachments/developerportal/deploy/mendix-cloud-deploy/deploy-to-cloud-node.png" alt="The App menu with the Deploy to Licensed Cloud Node option selected" width=50% class="image-border" >}} 
 

> 4. In the **Deploy to the cloud** dialog box, click **Deploy**. 

Author: > click **Deploy**.

Here we hit a major problem with naming within the product (assuming that all the UX elements are correctly described).
We are describing a two-stage process "create a package" and "deploy the package". But the UX keeps on using "deploy" to mean "create a package".

Not sure we can do anything to improve this - but if you can find a way, that would be great.
If we can't - then perhaps we need to create a naming paper and share it with the relevant PMs and ask them to look at making things more consistent.

Let me know if you decide to do this and need me to get involved in any political stuff.

5. You will see a dialog box with the following message: "Successfully started cloud deployment." To see the uploaded package, click **Show online** in the dialog box. 
 
##### 2.1.2.2 Creating a Deployment Package via the Developer Portal {#package-from-team-server} 
 

> You can also create a deployment package from the Team Server in the Developer Portal. To do this, follow these steps: 

Author: As above (I'm sure you can come up with something better!)
```suggestion
You can also create a deployment package through the Developer Portal, using a model committed to the Team Server. To do this, follow these steps: 
```

 
1. Go to the [Developer Portal](http://sprintr.home.mendix.com). 
1. Open your app's **Environments** page. 
1. Click **Create Deployment Package**. 
1. Select your desired branch and revision and click **Next**. 
1. Give the build a version number and click **Build this revision**. 
 
#### 2.1.3 Deploying the App to an Environment {#deploy-the-app-to-an-environment} 
 

> The previous steps explained how to deploy a deployment package to Mendix Cloud, but the actual app is not running yet. To deploy a deployment package to a node environment where you can run your app, follow these steps: 

Author: Do we need "actual app"?
Here we are struggling with a lack of distinction between _the app_=code and _the app_=the things which end-users interact with. I'm not sure that "actual" helps in this distinction. Perhaps there is another way of making it?

 
1. Go to the [Developer Portal](http://sprintr.home.mendix.com). 
2. Open your app. 
3. Go to **Environments**. 
4. In the **Deployment Package Repository**, choose your preferred deployment package and click **Deploy** ({{% icon name="deploy" %}}). 
 
    {{< figure src="/attachments/developerportal/deploy/mendix-cloud-deploy/deploy-package.png" class="image-border" alt="" >}} 
 
5. Select the environment to which you want to deploy the package. 
6. Click **Transport**. 
 
    {{< figure src="/attachments/developerportal/deploy/mendix-cloud-deploy/transport-to-flex.png" alt="The Transport menu" class="image-border" >}} 
 
7. If prompted, clean the environment. 

> 8. Configure the [constants](/refguide/constants/) (if necessary) by clicking **Constants** and **Edit constants value**. This can also be done at a later stage in the settings. 

Author: > This can also be done at a later stage in the settings.

I know what is meant, but I'm not sure it is

1. clear to the customer
2. good advice to give as I think constants are only picked up when the app is restarted? (see https://docs.mendix.com/refguide/constants/#42-exposed-to-client)

So perhaps we ought not to give this alternative?

> 9. Configure the [scheduled events](/refguide/scheduled-events/) (if necessary) by clicking **Scheduled Events**. 

Author: Perhaps "any" (or just nothing?)
```suggestion
9. Configure any [scheduled events](/refguide/scheduled-events/) (if necessary) by clicking **Scheduled Events**. 
```

Author: And perhaps that makes "(if necessary)" redundant too?

10. Click **Continue**. 
11. Click **Start application**.  
12. If prompted, click **Synchronize database**. 
 
The app is now deployed. You can configure the administrative account. 
 
### 2.2 Deploying a Free App {#deploy-free-app} 

>  

Author: Explain this is a single stage process, compared to the licensed process above.

#### 2.2.1 Prerequisites 
 
Before starting the process for deploying a Free App, make sure to complete these prerequisites: 
 
* You have created an app 
* As with licensed apps, the uncompressed size of the deployment package must not exceed 1 GB 
 
#### 2.2.2 Deploying the App 
 
With a Free App, you can deploy your app to Mendix Cloud from Studio Pro by using one of the following methods: 
 

> * In the top bar of Studio Pro, click **Run** > **Publish**. This automatically deploys your app to a Free App environment. 

Author: I believe this also works for licensed apps where a target environment has been defined. So we need to think about explaining this use case as well?
![image](https://github.com/mendix/docs/assets/35492184/0203746c-88bc-444b-9677-c74b26bc0603)

Reviewer: I'm unable to replicate this. I can set it up from the Developer Portal, but Publish and Run > Publish are greyed out for my licensed apps. Do you know if https://docs.mendix.com/developerportal/deploy/studio-deployment-settings/ has been replaced by https://docs.mendix.com/refguide/using-version-control-in-studio-pro/#83-choosing-a-specific-development-line-and-revision?

Author: I'm afraid I don't know. This was always a bit of a grey area. I think that the SAP and Private Cloud teams made sure that it worked for them, but I don't know about licensed apps.
I think you will have to contact the developers and ask how it should be working. It could be that our documentation is out of date - or the UX is misleading and wasn't changed after the withdrawal of Studio?

Reviewer: Thanks. I'll put it on my to-do list to follow up with them about this.

 
    {{< figure src="/attachments/developerportal/deploy/mendix-cloud-deploy/runapp2.png" alt="The Run menu with the Publish option selected"  width=60% >}} 
 
* Alternatively, in the top bar of Studio Pro, click **Publish**. 
 
    {{< figure src="/attachments/developerportal/deploy/mendix-cloud-deploy/runapp.png" alt="The Publish button" width=20% >}} 
 

> ## 3 Status of Deployment 

Author: This looks more like a **Troubleshooting** type section. Not sure the section title is that helpful. 
It should probably go wherever the **Deploying an App to Mendix Cloud** ends up. But in this document I would probably move it to the end.

| XL20        | 16 GiB  | 4        | 16 GiB | 2       | 80 GiB     | 160 GiB      | 
| XXL20       | 32 GiB  | 8        | 32 GiB | 2       | 160 GiB    | 320 GiB      | 
| Strato      | 2 GiB   | 0.5      | 1 GiB  | 0.5     | 5 GiB      | 20 GiB       | 
| Meso        | 2 GiB   | 0.5      | 8 GiB  | 2       | 20 GiB     | 20 GiB       | 
| Iono        | 8 GiB   | 2        | 8 GiB  | 2       | 20 GiB     | 80 GiB       | 
| Magneto     | 16 GiB  | 4        | 16 GiB | 4       | 80 GiB     | 320 GiB      | 
| S           | 1 GiB   | 0.5      | 1 GiB  | 0.5     | 5 GiB      | 5 GiB        | 
| M           | 2 GiB   | 0.5      | 2 GiB  | 1       | 10 GiB     | 10 GiB       | 
| L           | 4 GiB   | 1        | 4 GiB  | 2       | 20 GiB     | 20 GiB       | 
| XL          | 8 GiB   | 2        | 8 GiB  | 2       | 40 GiB     | 40 GiB       | 
| XXL         | 16 GiB  | 4        | 16 GiB | 4       | 80 GiB     | 80 GiB       | 
 
{{% alert color="info" %}} 
You can have a pack using a fraction of a vCPU. This is because several environments can run, completely isolated from each other, on a single virtual machine. 
{{% /alert %}} 
 
### 4.3 Additional Resources{#additional-resources} 
 

> For Premium customers using a Premium resource pack, there are additional features available for apps deployed to Mendix Cloud. There are also additional ways to deploy Mendix Apps. 

Author: ```suggestion
For Premium customers using a Premium resource pack, there are additional features available for apps deployed to Mendix Cloud. There are also additional ways to deploy Mendix apps. 
```
